lib/gtfs/cache_file.ex

```diff
@@ -0,0 +1,118 @@
+defmodule Gtfs.CacheFile do
+  @moduledoc """
+  This module holds the logic for loading and saving cache files
+  in dev and test environments with the goal of decreasing startup
+  time so devs can work more quickly.
+
+  If GTFS data changes, the cache files should be deleted/removed
+  to allowing caching of the changes.
+  """
+  require Logger
+
+  @directory Path.join([File.cwd!(), "priv/gtfs_cache"])
+
+  @doc """
+  The application should use the file to load state for GTFS
+  if the the envs are :dev or :test.
+
+    iex> Mix.env
+    :test
+    iex> CacheFile.should_use_file?
+    true
+  """
+  @spec should_use_file?() :: boolean
+  def should_use_file? do
+    Application.get_env(:skate, __MODULE__)[:cache_filename] != nil
```

Comment:
We're doing two checks for `nil`. One here to decide whether to attempt the cache, and one when we try to get the filename. There could be bugs if the two checks become slightly different. Is there a way to just have a single `get_filename_if_we_should_cache() :: String.t | nil` for both purposes, so we only need to make one decision?

It's a small thing and I think it'd require a little bit more of a refactor, so this probably isn't worth addressing, but if you see an easy way to do it it might help.

Reply:
Removed the other case.